### PR TITLE
feat: allows setting EUS repos with a flag

### PR DESCRIPTION
**Pull request by faiq**
**What problem does this PR solve?**:
Allows users to optionally disable using EUS repos for redhat 8.6 and 8.8

Also sets owner permissions to solve : https://jira.nutanix.com/browse/NCN-101832

**Which issue(s) does this PR fix?**:

* https://jira.d2iq.com/browse/D2IQ-NUMBER

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: bundles/redhat8.6/bundle.sh.gotmpl [hunk 1]
```diff
@@ -45,9 +45,19 @@ subscription::defer_unregister() {
 subscription-manager release --set=8.6
 subscription-manager refresh
 subscription::defer_unregister
-subscription-manager repos --enable rhel-8-for-x86_64-baseos-eus-rpms
-subscription-manager repos --enable codeready-builder-for-rhel-8-x86_64-rpms
-subscription-manager repos --enable rhel-8-for-x86_64-appstream-rpms
+
+ENABLED_REPOS="codeready-builder-for-rhel-8-x86_64-rpms,rhel-8-for-x86_64-appstream-rpms,rhel-8-for-x86_64-baseos-rpms"
```
Comment (dkoshkin) [inline bundles/redhat8.6/bundle.sh.gotmpl:49]: WDYT about putting this inside the `else`, closer to where they are enabled?

File: bundles/redhat8.6/bundle.sh.gotmpl [hunk 2]
```diff
@@ -57,9 +67,13 @@ pushd "${TMP_DIR}"
 repoquery --archlist=x86_64,noarch --resolve --requires --recursive  $(< packages.txt) | grep -v *.i686  >> reqs.txt
 sed -i 1d reqs.txt # we need to get rid of the first line
 #shellcheck disable=SC2046
-yumdownloader --archlist=x86_64,noarch --setopt=skip_missing_names_on_install=False -x \*i686 $(< reqs.txt)
+yumdownloader --archlist=x86_64,noarch \
+  --setopt=skip_missing_names_on_install=False -x \*i686 $(< reqs.txt)
 #shellcheck disable=SC2046
-yumdownloader --setopt=skip_missing_names_on_install=False -x \*i686 --archlist=x86_64,noarch --resolve --disablerepo=*  --enablerepo=kubernetes,rhel-8-for-x86_64-baseos-eus-rpms,codeready-builder-for-rhel-8-x86_64-rpms,rhel-8-for-x86_64-appstream-rpms $(< packages.txt)
+yumdownloader --setopt=skip_missing_names_on_install=False -x \*i686 --archlist=x86_64,noarch \
+  --resolve --disablerepo=* --enablerepo="kubernetes,${ENABLED_REPOS}" \
```
Comment (dkoshkin) [inline bundles/redhat8.6/bundle.sh.gotmpl:74]: WDYT about putting `kubernetes` in `ENABLED_REPOS`?

Comment (supershal) [reply bundles/redhat8.6/bundle.sh.gotmpl:74]: +1

File: bundles/redhat8.6/bundle.sh.gotmpl (hunk 1 above)
Comment (faiq) [reply bundles/redhat8.6/bundle.sh.gotmpl:49]: I think its fine to leave it here, these are the default values.

File: bundles/redhat8.8/packages.txt.gotmpl [hunk 3]
```diff
@@ -31,8 +31,8 @@ make
 libseccomp
 nfs-utils
 iproute-tc
-kernel-headers-4.18.0-477.58.1.el8_8
-kernel-devel-4.18.0-477.58.1.el8_8
+kernel-headers
```
Comment (supershal) [inline bundles/redhat8.8/packages.txt.gotmpl:34]: we no longer need to pin the kernel header versions?

If thats the case, we need similar fix for redhat-8.6

Comment (supershal) [reply bundles/redhat8.8/packages.txt.gotmpl:34]: in that case, should we create template variable for EUS support and toggle version accordingly?

Comment (faiq) [reply bundles/redhat8.8/packages.txt.gotmpl:34]: i think thats a bit too overkill for this. 

I think this is something that very few customers run into and the kernel version requirements should be documented. Whatever we set for our tests can be used just for that.

**Review by supershal — APPROVED** [review]

**Review by dkoshkin — APPROVED** [review]

File: bundles/redhat8.8/packages.txt.gotmpl [hunk 4]
```diff
@@ -36,3 +36,5 @@ kernel-devel
 gssproxy
 libverto-module-base
 libverto
+kernel-headers
+kernel-devel
```
Comment (dkoshkin) [inline bundles/redhat8.8/packages.txt.gotmpl:40]: These are already set on lines 34 and 35

File: bundles/redhat8.6/packages.txt.gotmpl [hunk 5]
```diff
@@ -31,7 +31,7 @@ elfutils-libelf-devel
 libseccomp
 nfs-utils
 iproute-tc
-kernel-headers
-kernel-devel
+kernel-headers-4.18.0-372.103.1.el8_6
+kernel-devel-4.18.0-372.103.1.el8_6
```
Comment (dkoshkin) [inline bundles/redhat8.6/packages.txt.gotmpl:35]: :/ it always tries to download the 8.8 version

https://github.com/mesosphere/konvoy-image-builder/actions/runs/10254800579/job/28370433396?pr=1133#step:9:2136

Comment (faiq) [reply bundles/redhat8.6/packages.txt.gotmpl:35]: thats a link to a rhel 8.8 run. it seems like we have to pin it in redhat 8.8 as well.

Comment (dkoshkin) [reply bundles/redhat8.6/packages.txt.gotmpl:35]: 🤦‍♂️ ok at least that explains it, I was starting to look why we would see this issue in 8.6 but not 8.8 now that 8.10 is out.

Comment (dkoshkin) [reply bundles/redhat8.6/packages.txt.gotmpl:35]: Before pinning, can you please try `unregistering` as described in https://access.redhat.com/solutions/238533

I wonder if `subscription-manager register` automatically subscribes to these repos.
```
subscription-manager repos --disable=rhel-8-for-x86_64-baseos-rpms --disable=rhel-8-for-x86_64-appstream-rpms
```

File: bundles/redhat8.8/packages.txt.gotmpl (hunk 4 above)
Comment (faiq) [reply bundles/redhat8.8/packages.txt.gotmpl:40]: :man_facepalming: nice catch thank you

File: bundles/redhat8.6/packages.txt.gotmpl (hunk 5 above)
Comment (dlipovetsky) [reply bundles/redhat8.6/packages.txt.gotmpl]: I didn't see any mention of `unregister` in the above article.

Did you mean the part that talks about `subscription-manager release --unset`?

File: bundles/redhat8.6/bundle.sh.gotmpl [hunk 6]
```diff
@@ -49,6 +49,7 @@ subscription::defer_unregister
 ENABLED_REPOS="kubernetes,codeready-builder-for-rhel-8-x86_64-rpms,rhel-8-for-x86_64-appstream-rpms,rhel-8-for-x86_64-baseos-rpms"
 EUS_REPOS=${EUS_REPOS:-""}
 if [[ -n "${EUS_REPOS}" ]]; then
+  subscription-manager repos --disable=rhel-8-for-x86_64-baseos-rpms --disable=rhel-8-for-x86_64-appstream-rpms #disable baseos repos
```
Comment (dlipovetsky) [inline bundles/redhat8.6/bundle.sh.gotmpl:52]: nit: Can you please explain what this does in more details, and make it more prominent? 

(Does this remove the repos for the "current" RHEL release? :thinking:)

Comment (faiq) [reply bundles/redhat8.6/bundle.sh.gotmpl]: i added a note that explains what it does and why.

File: bundles/redhat8.6/bundle.sh.gotmpl [hunk 7]
```diff
@@ -45,25 +45,42 @@ subscription::defer_unregister() {
...
 subscription-manager refresh
 subscription::defer_unregister
-subscription-manager repos --enable rhel-8-for-x86_64-baseos-eus-rpms
-subscription-manager repos --enable codeready-builder-for-rhel-8-x86_64-rpms
-subscription-manager repos --enable rhel-8-for-x86_64-appstream-rpms
+
+ENABLED_REPOS="kubernetes,codeready-builder-for-rhel-8-x86_64-rpms,rhel-8-for-x86_64-appstream-rpms,rhel-8-for-x86_64-baseos-rpms"
+EUS_REPOS=${EUS_REPOS:-""}
+if [[ -n "${EUS_REPOS}" ]]; then
+  #disables the standart repositories which should not be enabled when using EUS
```
Comment (dlipovetsky) [inline bundles/redhat8.6/bundle.sh.gotmpl:52]: nit
```suggestion
  #disables the standard repositories which should not be enabled when using EUS
```

File: bundles/redhat8.8/bundle.sh.gotmpl [hunk 8]
```diff
@@ -45,18 +45,26 @@ subscription::defer_unregister() {
...
-subscription-manager repos --enable rhel-8-for-x86_64-baseos-rpms
 
-# EUS rpms used to fetch kernel-headers for redhat 8.8
-# if you don't need kernel headers package for NVIDIA Gpu support, remove this line.
-# and remove kernel-headers-4.18.0-477.58.1.el8_8 from packages.txt.gotmpl
-subscription-manager repos --enable rhel-8-for-x86_64-baseos-eus-rpms
+ENABLED_REPOS="kubernetes,codeready-builder-for-rhel-8-x86_64-rpms,rhel-8-for-x86_64-appstream-rpms,rhel-8-for-x86_64-baseos-rpms"
+EUS_REPOS=${EUS_REPOS:-""}
+if [[ -n "${EUS_REPOS}" ]]; then
+  #disables the standart repositories which should not be enabled when using EUS
```
Comment (dlipovetsky) [inline bundles/redhat8.8/bundle.sh.gotmpl:52]: nit
```suggestion
  #disables the standard repositories which should not be enabled when using EUS
```

**Review by dlipovetsky — APPROVED** [review]

File: bundles/redhat8.6/bundle.sh.gotmpl (hunk 7 above)
Comment (faiq) [reply bundles/redhat8.6/bundle.sh.gotmpl:52]: thank you !!

File: bundles/redhat8.6/packages.txt.gotmpl (hunk 5 above)
Comment (faiq) [reply bundles/redhat8.6/packages.txt.gotmpl]: I added the disable line, and i think it gets us closer to a package bundle specifically for a EUS point version eg 8.8 but it doesn't necessarily solve the issue of matching kernel headers versions to the kernel.

https://github.com/mesosphere/konvoy-image-builder/pull/1136

should solve disabling it for users who don't need it and some new docs tracked by https://jira.nutanix.com/browse/NCN-101854 can explain what to put for the version and how to get it.

**Pull request merged**